src/phoebus_driver.cpp

```diff
@@ -1014,11 +1017,211 @@ TaskListStatus PhoebusDriver::MonteCarloStep() {
+
+  IndexRange ib = rc->GetBoundsI(IndexDomain::interior);
+  IndexRange jb = rc->GetBoundsJ(IndexDomain::interior);
+  IndexRange kb = rc->GetBoundsK(IndexDomain::interior);
+
+  auto eos = pmb->packages.Get("eos")->Param<Microphysics::EOS::EOS>("d.EOS");
+  parthenon::par_for(
+      DEFAULT_LOOP_PATTERN, "UserWorkBeforeOutput::H5", DevExecSpace(), kb.s, kb.e, jb.s,
+      jb.e, ib.s, ib.e, KOKKOS_LAMBDA(const int k, const int j, const int i) {
+        Real lambda[2];
+        if (v.Contains(0, p::ye())) {
+          lambda[0] = v(0, p::ye(), k, j, i);
+        }
+        const Real s = eos.EntropyFromDensityTemperature(
+            v(0, p::density(), k, j, i), v(0, p::temperature(), k, j, i), lambda);
+        const Real p = eos.PressureFromDensityTemperature(
+            v(0, p::density(), k, j, i), v(0, p::temperature(), k, j, i), lambda);
+        const Real bmod = eos.BulkModulusFromDensityTemperature(
+            v(0, p::density(), k, j, i), v(0, p::temperature(), k, j, i), lambda);
+        const Real sie = eos.InternalEnergyFromDensityTemperature(
+            v(0, p::density(), k, j, i), v(0, p::temperature(), k, j, i), lambda);
+        const Real h = sie + p / v(0, p::density(), k, j, i) + 1;
+        const Real cs = std::sqrt(bmod / v(0, p::density(), k, j, i) / h);
+        Real divv;
+        Real gam[3][3];
+        Real gam1[3][3];
+        Real gam2[3][3];
+        Real gam3[3][3];
+        Real gam12[3][3];
+        Real gam13[3][3];
+        Real gam23[3][3];
+        Real gam123[3][3];
+        const Real vp[3] = {v(0, p::velocity(0), k, j, i), v(0, p::velocity(1), k, j, i),
+                            v(0, p::velocity(2), k, j, i)};
+        const Real vp1[3] = {v(0, p::velocity(0), k, j, i - 1),
+                             v(0, p::velocity(1), k, j, i - 1),
+                             v(0, p::velocity(2), k, j, i - 1)};
+        geom.Metric(CellLocation::Cent, 0, k, j, i, gam);
+        geom.Metric(CellLocation::Cent, 0, k, j, i - 1, gam1);
+        Real gdet = geom.DetGamma(CellLocation::Cent, 0, k, j, i);
+        const Real W = phoebus::GetLorentzFactor(vp, gam);
+        const Real W1 = phoebus::GetLorentzFactor(vp1, gam1);
+
+        if (ndim == 1) {
+          divv = 1 / gdet *
+                 (geom.DetGamma(CellLocation::Cent, 0, k, j, i) *
+                      v(0, p::velocity(0), k, j, i) / W -
+                  geom.DetGamma(CellLocation::Cent, 0, k, j, i - 1) *
+                      v(0, p::velocity(0), k, j, i - 1) / W1) /
+                 coords.CellWidthFA(X1DIR, k, j, i);
+        }
+
+        if (ndim == 2) {
+          Real vp2[3] = {v(0, p::velocity(0), k, j - 1, i),
+                         v(0, p::velocity(1), k, j - 1, i),
+                         v(0, p::velocity(2), k, j - 1, i)};
+          Real vp12[3] = {v(0, p::velocity(0), k, j - 1, i - 1),
+                          v(0, p::velocity(1), k, j - 1, i - 1),
+                          v(0, p::velocity(2), k, j - 1, i - 1)};
```

Comment:
Why is this one on the diagonal? I don't think you need that. In the divergence formula, we have
$v^i_{;i} =\frac{1}{ \sqrt{\gamma} }\partial_i (\sqrt{\gamma}v^i)$
so you only need the derivatives in the code coordinate directions, not mixed derivatives.